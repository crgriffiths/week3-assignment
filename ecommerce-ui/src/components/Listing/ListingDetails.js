import React from 'react';
import PropTypes from 'prop-types';

class ListingDetails extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div id={`listing-${this.props.index}`} className="listing-details">
        <div>
          <p>{this.props.houseType} {this.props.location}</p>
        </div>
        <div>
          <h2>{this.props.title}</h2>
        </div>
        <div>
          <p>${this.props.payment.cost} {this.props.payment.description.length > 0 ? `- ${this.props.payment.description}` : ""}</p>
        </div>
        <div>
          <p><span>{this.props.rating.stars} out of 5 ({this.props.rating.reviews}){this.props.host.isSuperhost ? "  \u2022 Superhost" : ""}</span></p>
        </div>
        <div>
          <button onClick={()=> this.props.onAddToCart(this.props.index)}>Reserve</button>
        </div>
      </div>
    )
  }
}

ListingDetails.propTypes = {
  index: PropTypes.number,
  houseType: PropTypes.string,
  location: PropTypes.string,
  title: PropTypes.string,
  payment: PropTypes.object,
  host: PropTypes.object,
  rating: PropTypes.object,
  onAddToCart: PropTypes.func
};

export default ListingDetails;