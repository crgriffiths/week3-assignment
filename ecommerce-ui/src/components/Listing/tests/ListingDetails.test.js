import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import ListingDetails from '../ListingDetails';

configure({adapter: new Adapter()});
describe('Listing Details', () => {
  it('Should render listing details', () => {
    const component = shallow(<ListingDetails 
      index={0}
      houseType="Apartment"
      location="New York"
      title="Centrally-located Manhattan studio"
      payment={
        {
          "cost": 105,
          "description": "Free cancellation"
        }
      }
      host={
        {
          "name": "Robert",
          "isSuperhost": true
        }
      }
      rating={
        {
          "stars": 3,
          "reviews": 133
        }
      }
    />);
    expect(component).toMatchSnapshot();
  });
})
