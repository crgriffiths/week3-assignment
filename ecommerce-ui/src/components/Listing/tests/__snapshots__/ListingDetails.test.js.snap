// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing Details Should render listing details 1`] = `ShallowWrapper {}`;
